ipsec: improve tuning documentation (#786)
opnsense/docs: source/manual/vpnet.rst

making sure that traffic is processed in parallel. The distribution mechanism (before any OS processing) making this possible often hashes
the IP/TCP/UDP 5-tuple to determine on which CPU a flow should be handled. Proper distribution therefore requires proper uniqueness
of packet headers.

When any tunneling takes place, as is the case for IPsec with ESP flows, many different traffic flows may be
encapsulated inside a single IP header, reducing uniqueness of traffic and therefore the ability to distribute work. Unless
NICs or virtualized queues can distribute work on ESP packets, which is rarely the case, OPNsense firewalls can often only
receive such tunneled flows on a single CPU, because the outer IP header is often the same.

If you have a simple, single-tunnel, single-flow setup (two flows for in-out), you may benefit from using async crypto to dispatch cryptographic work
to multiple processors. However, there is no guarantee this will speed up work as it is dependent on multiple factors.

In order to do so, add or change the following tunable in :menuselection:`System --> Settings --> Tunables`:

.. Note::

    :code:`net.inet.ipsec.async_crypto` = **1**

> As soon as an IPsec packet is received, decryption must happen first before anything meaningful can be done with it. This process

Reviewer: ```suggestion
As soon as an IPsec packet is received, decryption must be handled first before anything meaningful can be done with the actual data. This process
```
